modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/CompanyIndexFactory.java

```diff
@@ -235,17 +233,23 @@ public void replaceCurrentIndexWithNextIndex(long companyId)
 
 		IndicesClient indicesClient = restHighLevelClient.indices();
 
-		if (_crossClusterReplicationHelper != null) {
-			_crossClusterReplicationHelper.unfollow(removeIndex);
+		CrossClusterReplicationHelper crossClusterReplicationHelper =
+			_crossClusterReplicationHelperSnapshot.get();
+
+		if (crossClusterReplicationHelper != null) {
+			crossClusterReplicationHelper.unfollow(removeIndex);
 		}
 
 		indicesClient.updateAliases(
 			indicesAliasesRequest, RequestOptions.DEFAULT);
 
 		_companyLocalService.updateIndexNames(companyId, indexNameNext, null);
 
-		if (_crossClusterReplicationHelper != null) {
-			_crossClusterReplicationHelper.follow(indexNameNext);
+		crossClusterReplicationHelper =
+			_crossClusterReplicationHelperSnapshot.get();
```

Comment:
@W2jx Why do you get again here?

Reply:
Hi @dantewang , Because the variable crossClusterReplicationHelper may be null, in the original logic, the variable _crossClusterReplicationHelper is dynamically changed by osgi, but in the modified code, I can only get the latest situation of the variable crossClusterReplicationHelper by re-get it.